dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,569 @@
+import com.dotmarketing.util.WebKeys.Relationship.RELATIONSHIP_CARDINALITY;
+import com.dotmarketing.util.json.JSONObject;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import com.tngtech.junit.dataprovider.format.DataProviderTestNameFormatter;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Class used to validate ES mapping for fields is applied correctly considering fields types and dynamic mapping defined in `es-content-mapping.json`
+ * @author nollymar
+ */
+@RunWith(DataProviderRunner.class)
+public class ESMappingUtilHelperTest {
+
+    private static User user;
+    private static CategoryAPI categoryAPI;
+    private static ContentletAPI contentletAPI;
+    private static ContentletIndexAPI contentletIndexAPI;
+    private static ContentTypeAPI contentTypeAPI;
+    private static ESMappingAPIImpl esMappingAPI;
+    private static FieldAPI fieldAPI;
+    private static RelationshipAPI relationshipAPI;
+    private static Language language;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        user = APILocator.systemUser();
+        categoryAPI = APILocator.getCategoryAPI();
+        contentletAPI = APILocator.getContentletAPI();
+        contentletIndexAPI = APILocator.getContentletIndexAPI();
+        contentTypeAPI = APILocator.getContentTypeAPI(user);
+        esMappingAPI = new ESMappingAPIImpl();
+        fieldAPI = APILocator.getContentTypeFieldAPI();
+        relationshipAPI = APILocator.getRelationshipAPI();
+        language = APILocator.getLanguageAPI().getDefaultLanguage();
+    }
+
+    public static class PlusTestNameFormatter implements DataProviderTestNameFormatter {
+        @Override
+        public String format(Method testMethod, int invocationIndex, List<Object> arguments) {
+            return String.format("Test Name: %s. Mapping Name: %s", testMethod.getName(), arguments.get(0));
+        }
+    }
+
+    @DataProvider(formatter = PlusTestNameFormatter.class)
+    public static Object[][] dataProviderValidateEventMapping(){
+        return new Object[][] {
+                {"template_1", new String[]{"host.hostname_dotraw"}, "keyword"},
+                {"textmapping", new String[]{"host.hostname"}, "keyword"},
+                {"strings_as_dates", new String[]{"calendarevent.originalstartdate",
+                        "calendarevent.recurrencestart", "calendarevent.recurrenceend"}, "date"},
+                {"permissions", new String[]{"permissions"}, "text"},
```

Comment:
Codacy found an issue: [The String literal "permissions" appears 4 times in this file; the first occurrence is on line 109](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)